flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/validate/FunctionCatalog.scala

```diff
@@ -47,13 +52,50 @@ class FunctionCatalog {
     sqlFunctions += sqlFunction
   }
 
+  /** Register multiple sql functions at one time. The functions has the same name. **/
+  def registerSqlFunctions(functions: Seq[SqlFunction]): Unit = {
+    if (functions.nonEmpty) {
+      sqlFunctions --= sqlFunctions.filter(_.getName == functions.head.getName)
+      sqlFunctions ++= functions
+    }
+  }
+
   def getSqlOperatorTable: SqlOperatorTable =
     ChainedSqlOperatorTable.of(
       new BasicOperatorTable(),
       new ListSqlOperatorTable(sqlFunctions)
     )
 
   /**
+    * Lookup table function and create an TableFunctionCall if we find a match.
+    */
+  def lookupTableFunction[T](name: String, children: Seq[Expression]): TableFunctionCall[T] = {
+    val funcClass = functionBuilders
+      .getOrElse(name.toLowerCase, throw ValidationException(s"Undefined function: $name"))
+    funcClass match {
+      // user-defined table function call
+      case tf if classOf[TableFunction[T]].isAssignableFrom(tf) =>
+        Try(UserDefinedFunctionUtils.instantiate(tf.asInstanceOf[Class[TableFunction[T]]])) match {
+          case Success(tableFunction) => {
+            val clazz: Type = tableFunction.getClass.getGenericSuperclass
+            val generic = clazz match {
+              case cls: ParameterizedType => cls.getActualTypeArguments.toSeq.head
+              case _ => throw new TableException(
+                "New TableFunction classes need to inherit from TableFunction class," +
+                  " and statement the generic type.")
+            }
+            implicit val typeInfo: TypeInformation[T] = TypeExtractor.createTypeInfo(generic)
+              .asInstanceOf[TypeInformation[T]]
+            TableFunctionCall(tableFunction, children, None)
+          }
+          case Failure(e) => throw ValidationException(e.getMessage)
+        }
+      case _ =>
+        throw ValidationException("Unsupported table function.")
```

Comment:
I think this exception message could be improved. It is throw if the registered method does not implement the `TableFunction` interface.